Source/objects.cpp

```diff
@@ -4645,13 +4542,13 @@ bool OperateFountains(int pnum, int i)
 		if (pnum == myplr)
 			NetSendCmdParam1(false, CMD_OPERATEOBJ, i);
 		break;
-	case OBJ_TEARFTN:
+	case OBJ_TEARFTN: {
 		if (object[i]._oSelFlag == 0)
 			break;
-		prev = -1;
-		add = -1;
-		done = false;
-		cnt = 0;
+		int prev = -1;
+		int add = -1;
+		bool done = false;
+		int cnt = 0;

```

Comment:
Move these above the while loop on 4559? Looking at that loop it seems like `done` is unnecessary and `cnt` only exists to make it run twice... I've created https://github.com/diasurgical/devilutionX/issues/2278 for that refactoring opportunity.